### PR TITLE
Bf refactor/percentage visitor wins

### PR DESCRIPTION
**Pull request by bflanagan138**
Refactor percentage_visitor_wins, percentage_home_wins, percentage_ties methods
Rewrite total_games_by_coaches_by_season and coach_stats_by_season helper methods
Testing 98.56% passing

### CODE REVIEW

**Review by gabrielcnunez — APPROVED** [review]
The refactors on the wins and tie percentage methods really helps with readability, great job!

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -35,43 +35,33 @@ def lowest_total_score
...
 
   # Original method from Iteration 2
   # Recommend combining percentage_ties, percentage_home_wins, percentage_visitor_wins methods using mixins or ?
   def percentage_home_wins
-    home_wins = 0
-   @games.each do |row|
-      if row[:away_goals] < row[:home_goals]
-        home_wins += 1
-      end
+    home_wins = @games.count do |row|
```
Comment (josephhilby) [inline lib/stat_tracker.rb:43]: Awesome use of .count to reduce some lines!

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -243,10 +233,11 @@ def total_games_by_coaches_by_season(season)
       if game_id_by_season.include?(game[:game_id]) 
         @games_by_coaches_by_season[game[:head_coach]] += [game]
       end
+      # binding.pry
```
Comment (josephhilby) [inline lib/stat_tracker.rb]: Removed old # binding.pry on line 236.

**Review by josephhilby — APPROVED** [review]
Looks good. That should complete all of our tests!

**Pull request merged**